forte/data/data_store.py

```diff
@@ -313,52 +399,82 @@ def delete_entry(self, tid: int):
-        # _delete_entry_by_loc.
+        # `entry_type`, `type_id`. If it is an Annotation-Like entry, we obtain
+        # `begin` and `end` indices and use bisection to locate the `index_id`.
+        # If it is not, we obtain the 'index_id' directly from `__entry_dict`.
+        # We remove the entry from `__entry_dict` when its information is
+        # retrieved. Then, the `type_id` and the `index_id` are passed to
+        # `_delete_entry_by_loc`.
         raise NotImplementedError
 
     def _delete_entry_by_loc(self, type_id: int, index_id: int):
         r"""It removes an entry of `index_id` by taking both the `type_id`
         and `index_id`. Called by `delete_entry()`.
 
         Args:
-            type_id (int): The index of the sortedlist in `self.__elements`.
-            index_id (int): The index of the entry in the sortedlist.
+            type_id (int): The index of list in `self.__elements`.
+            index_id (int): The index of the entry in the list.
         """
-        # We then remove the entry data from `entry_type` sortedlist.
+        # We then remove the entry data from `entry_type` list.
         raise NotImplementedError
 
     def get_entry(self, tid: int) -> Tuple[List, int, int]:
         r"""This function finds the entry with `tid`. It returns the entry,
-        its `type_id`, and the index in the `entry_type` sortedlist.
+        its `type_id`, and the index in the `entry_type` list.
 
         Args:
             tid (int): Unique id of the entry.
 
         Returns:
             The entry which `tid` corresponds to, its `type_id` and its index
-            in the `entry_type` sortedlist.
-
+            in the `entry_type` list.
         """
+        # If the entry is an annotation, bisect the annotation sortedlist
+        # to find the entry. May use LRU cache to optimize speed.
+        # Otherwise, use `index_id` to find the index of the entry.
         raise NotImplementedError
 
     def get(
         self, entry_type: Union[str, Type[EntryType]], **kwargs
     ) -> Iterator[List]:
-        r"""Fetch entries from the data store of type `entry_type`.
+        r"""This function fetches entries from the data store of
+        type `entry_type`.
 
         Args:
             entry_type: Fully qualified name of this annotation.
 
         Returns:
             An iterator of the entries matching the provided arguments.
-
         """
-        # We find the `type_id` according to `entry_type` and locate the
-        # sortedlist.
-        # We create an iterator to generate entries from the sortedlist.
+        # We find the `type_id` according to `entry_type` and locate the list.
+        # We create an iterator to generate entries from the list.
+        if not isinstance(entry_type, str):
+            # convert `entry_type` to fully qualified name
+            entry_type = get_full_module_name(entry_type)
         yield from self.__elements[self.__type_dict[entry_type]]
```

Comment:
This is actually oversimplifying things, did you consider the sub-classes?

Reply:
Do you mean user-defined classes? `get_full_module_name` now works for classes defined in ft/ftx like 'ftx.onto.ag_news.Description', 'ft.onto.base_ontology.Token', etc .

Reply:
No that one. Notice if you request a type, it is logical to also expect the sub-type of it also gets returned. 

We implemented similar stuff in data pack: https://github.com/asyml/forte/blob/master/forte/data/data_pack.py#L1365

So here we will need to use the parallel-iteration technique we discussed.

Reply:
Is the plan to leave the implementation to another PR?

Reply:
I had a brute force way to implement it and it is already in the code. I think we can optimize after the issue #583 is done.

Reply:
ok, once we merge this let's create an issue for that,

Reply:
yes sure